duckduckgo/iOS
# Improve app statistics

Task/Issue URL: https://app.asana.com/0/414235014887631/1109730748569886
Tech Design URL: https://app.asana.com/0/481882893211075/1109730748569882

**Description:**
Update the app to maintain two versions of the latest atb (this is the set_atb parameter sent in atb.js) to provide better statistics. One for search and one for app usage.

**Steps to test this PR**

**Test 1: Migration:**
1. Install and run a develop version of the app with an old atb and yesterday's retentionAtb (these values can be hardcoded in StatisticsUserDefaults)
1. Open Charles to watch requests
1. Now install this branch (revert any hardcoding code from step 1) over the top
1. Note that when the app starts an atb.js request is made with an additional parameter of `at` and the original atb for both `atb` and `set_atb`
https://duckduckgo.com/atb.js?test=1&at=app_use&atb=v157-0mn&set_atb=v157-0
1. Background the app and open again note that the atb request is fired again only this time set_atb contains the atb from the previous response e.g if the latest atb is v157-6
https://duckduckgo.com/atb.js?test=1&at=app_use&atb=v157-0mn&set_atb=v157-6
1. Perform a search, this will not contain the `at` parameter and will use the previously known searchRetentionAtb, which will be yesterday's value e.g v157-5
https://duckduckgo.com/atb.js?test=1&atb=v157-0mn&set_atb=v157-5
1. Perform another search and note that the search atb is up-to-date now too
https://duckduckgo.com/atb.js?test=1&atb=v157-0mn&set_atb=v157-6
1. Note that no exti or initial atb request (the ones with no `atb` or `set_atb`  params) are sent after step 1
`https://duckduckgo.com/exti/?test=1&atb=v157-6mn`
`https://duckduckgo.com/atb.js?test=1`

**Test 2: New installation:**
1. Install a fresh version of the app and watch requests in Charles
1. Note that the initial atb request (with no `atb` or `set_atb`  params) appears only once
`https://duckduckgo.com/atb.js?test=1`
1. Note that the `exti` request only appears once
`https://duckduckgo.com/exti/?test=1&atb=v157-6mn`

---
###### Internal references:
[Software Engineering Expectations](https://app.asana.com/0/59792373528535/199064865822552)
[Technical Design Template](https://app.asana.com/0/59792373528535/184709971311943)
